### PR TITLE
Structural Updates

### PR DESCRIPTION
**Pull request by smusali**
Updates:
* Adding `SemVer` and `CHANGELOG.md`
* Updating `CircleCI`
* ReStructuring
* Adding `user-agent` to `req.headers`
* Renaming `aws-cloudwatch`, `logdna_cloudwatch`, and `logdna_cloudwatchlogs` to `logdna-cloudwatch`

### CODE REVIEW

**Comment by smusali** [comment]
@dchai76 @andkon @vilyapilya, can you guys review the changes? Appreciated!

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Change Log
```
Comment (dchai76) [inline CHANGELOG.md]: It's usually one word: `Changelog`. https://en.wikipedia.org/wiki/Changelog

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+This file documents all notable changes in `LogDNA CloudWatch Lambda Function`. The release numbering uses [semantic versioning](http://semver.org).
+
+## v2.0.0 - Released on June 5, 2019
+* Releasing new `Lambda Function` written/implemented on `node.js`
```
Comment (dchai76) [inline CHANGELOG.md]: This read weird to me - I'm more used to changelogs having past tense. Like "Released new...", "Added retry"... etc. E.g. https://github.com/lodash/lodash/wiki/Changelog

File: logdna/index.js [hunk 3]
```diff
@@ -14,7 +14,10 @@ const REQUEST_RETRY_INTERVAL = parseInt(process.env.LOGDNA_REQUEST_RETRY_INTERVA
 
 // Get Configuration from Environment Variables
 const getConfig = () => {
-    let config = {};
+    const pkg = require('./package.json');
+    let config = {
+        UA: `${pkg.name}/${pkg.version}`
```
Comment (dchai76) [inline logdna/index.js]: What's wrong with spelling out `UserAgent`?

File: CHANGELOG.md (hunk 2 above)
Comment (smusali) [reply CHANGELOG.md]: Haha - done!

File: logdna/index.js (hunk 3 above)
Comment (smusali) [reply logdna/index.js]: Nothing wrong - I was just following what was implemented for `logdna-agent:v1.5.6`

Comment (smusali) [reply logdna/index.js]: but for clarity and readability, `UserAgent` is da best

**Comment by smusali** [comment]
@dchai76, can you do second round? Thanks in advance!

File: .circleci/config.yml [hunk 4]
```diff
@@ -5,23 +5,40 @@ jobs:
       - image: circleci/node:10
     steps:
       - checkout
+      - run:
+          name: Check Tagged Push
+          command: |
+            TAG_VERSION=$(git tag --sort=-creatordate | head -n 1)
```
Comment (dchai76) [inline .circleci/config.yml]: Random question - why this and not something like `git describe --tags --abbrev=0`

File: .circleci/config.yml [hunk 5]
```diff
@@ -31,12 +48,12 @@ workflows:
             branches:
               ignore: /.*/
             tags:
-              only: /^[0-9]{10}$/
+              only: /[0-9]+\.[0-9]+\.[0-9]+(\.[0-9]+)?/
```
Comment (dchai76) [inline .circleci/config.yml]: The frontend team has actually been sometimes publishing versions like `2.1.4a` but I'm fine ignoring that for now.

Comment (smusali) [reply .circleci/config.yml]: We are not gonna do that, most probably

File: .circleci/config.yml (hunk 4 above)
Comment (smusali) [reply .circleci/config.yml]: @dchai76, difference is easily visible when applied in our web repo

**Pull request merged**